src/components/shared/SourceView.js

```diff
@@ -143,6 +155,23 @@ export class SourceView extends React.PureComponent<SourceViewProps> {
     return this._computeAllLineNumbersMemoized(this._getSourceLines(), timings);
   }
 
+  /* This method is used by users of this component. */
+  /* eslint-disable-next-line react/no-unused-class-component-methods */
+  scrollToHotSpot(timingsForScrolling: LineTimings) {
+    const heaviestLine =
+      _mapGetKeyWithMaxValue(timingsForScrolling.totalLineHits) ??
+      _mapGetKeyWithMaxValue(this.props.timings.totalLineHits);
```

Comment:
Can you please explain quickly (in a comment) why we can have these 2 objects and why they're different? My understanding is that `timingsForScrolling.totalLineHits` is an empty map when there's no selected call node. But this isn't obvious, neither from here nor from `BottomBox`, and I had to look at the implementation in `line-timings.js` to understand this. Maybe returning `null` instead of empty maps from `getLineTimings` could help? I don't know.

Maybe it could be more explicit to check the `size` property here + add a comment to explain when this happens.

Reply:
I've added a long comment (in a new commit), but I think I'll keep the `??` construct.